src/ogma/features/styles.ts

```diff
@@ -589,38 +589,56 @@ export class StylesViz {
     return new StyleRules(state)[styleType];
   }
 
-  public initNodeColors(nodeRules: Array<IStyleRule<INodeStyle | IEdgeStyle>>) {
-    const nodeColorRules = this.getStyleRule(nodeRules, StyleType.COLOR);
+  public initNodeColors(nodeRules: Array<IStyleRule<INodeStyle | IEdgeStyle>>): void {
+    const nodeColorRules = this.getStyleRule(
+      nodeRules,
+      StyleType.COLOR
+    ) as LKStyleRule<INodeStyle>[];
     this.refreshNodeColors(nodeColorRules);
   }
 
   public initNodesIcons(nodeRules: Array<IStyleRule<INodeStyle | IEdgeStyle>>) {
-    const nodeIconsRules = this.getStyleRule(nodeRules, StyleType.ICON);
+    const nodeIconsRules = this.getStyleRule(
+      nodeRules,
+      StyleType.ICON
+    ) as LKStyleRule<INodeStyle>[];
     this.refreshNodeIcons(nodeIconsRules);
   }
 
   public initNodesSizes(nodeRules: Array<IStyleRule<INodeStyle | IEdgeStyle>>) {
-    const nodeSizeRules = this.getStyleRule(nodeRules, StyleType.SIZE);
+    const nodeSizeRules = this.getStyleRule(nodeRules, StyleType.SIZE) as LKStyleRule<INodeStyle>[];
     this.refreshNodeSize(nodeSizeRules);
   }
 
   public initNodesShapes(nodeRules: Array<IStyleRule<INodeStyle | IEdgeStyle>>) {
-    const nodeShapesRules = this.getStyleRule(nodeRules, StyleType.SHAPE);
+    const nodeShapesRules = this.getStyleRule(
+      nodeRules,
+      StyleType.SHAPE
+    ) as LKStyleRule<INodeStyle>[];
```

Comment:
you could avoid this type I think with a generic and condition typing. Something like:
getStyleRule<T>(rules: T): T extends NodeRules ? LKStyleRule<INodeStyle>[] : LKStyleRule<IEdgeStyle>[] {}
(it's pseudo code, it might not work, and just a suggestion, not mandatory)

Reply:
I think it will work but need to make StyleRules also generic and change it in multiple places.
Maybe in another PR if the need come up.